how-to/add_controller.rst

```diff
@@ -1,119 +1,120 @@
 correctly in the JAAS system. This document discusses how to bootstrap a juju controller
 such that it will work correctly in a JAAS system.
 
-In this tutorial we will be bootstrapping a new juju controller in AWS and adding it to
-JIMM.
+In this how-to we will show how to add Juju controllers deployed in both MicroK8s and LXD to 
+a JIMM controller.
 
 Prerequisites
 -------------
 
 For this tutorial you will need the following:
 
-- A valid registered domain (regardless of the registrar)
-- AWS credentials
 - Basic knowledge of juju
-- A subdomain registered with Route 53. To learn how to set that up, please follow :doc:`route53`. In this tutorial we will assume that you have registered the canonical.example.com subdomain - please replace this with the appropriate subdomain that you have registered with Route 53.
-- Admin access to a JIMM controller (see this tutorial). For this tutorial we will assume this JIMM is located at ``jimm.canonical.example.com``
+- A JIMM controller deployed in MicroK8s, see :doc:`the tutorial <../tutorial/deploy_jaas_microk8s>`.
+- Administrator permission on the JIMM controller, see ``todo``.
 
-Deploy controller
------------------
 
-1. First we will prepare some parameters for the new controller and export environment variables that we will use in this tutorial. 
+Prelude
+-------
 
-    The **controller name** is the name given to the controller both on the local system and within JIMM. For visibility this often includes the name of the JAAS system, the cloud, the cloud-region and some kind of unique identifier, for example ``jaas-aws-us-east-1-001``. 
+In order for a Juju controller to trust a JIMM controller, a specific config option must be set known
+as the ``login-token-refresh-url``.
 
-    The **cloud** is the cloud in which the controller is being bootstrapped. 
+This config option is set to a specific URL path on the JIMM controller that serves JIMM's public key.
+Requests from JIMM to the Juju controller are signed, and verified when they reach the Juju.
 
-    The **cloud region** is the region in which the controller is being bootstrapped. 
+MicroK8s Controller
+-------------------
 
-    The **DNS name** is the full DNS name that will be given to the controller, often it is wise to make the host name the same as the controller name within a particular domain. 
+Bootstrap a new Juju controller in MicroK8s. We will name this controller ``workload-microk8s`` as it will be running our workloads
+as opposed to our original controller which only deploys JAAS.
 
-    The **Candid URL** is the URL of the candid server that is providing the centralised identity service for the JAAS system. 
+.. code:: bash
 
-    The **JIMM URL** is the URL of the JIMM system providing the JAAS service.
+    juju bootstrap microk8s workload-microk8s --config login-token-refresh-url=http://jimm-endpoints.jimm.svc.cluster.local:8080/.well-known/jwks.json
```

Comment:
should we add an explanation where this DNS comes from?

Reply:
Added a note.